chain-signatures/node/src/cli.rs

```diff
@@ -161,6 +161,40 @@ impl Cli {
     }
 }
 
+fn spinup_indexer(
+    options: &indexer::Options,
+    mpc_contract_id: &AccountId,
+    account_id: &AccountId,
+    sign_queue: &Arc<RwLock<SignQueue>>,
+    gcp: &GcpService,
+) -> std::thread::JoinHandle<()> {
+    let options = options.clone();
+    let mpc_contract_id = mpc_contract_id.clone();
+    let account_id = account_id.clone();
+    let sign_queue = sign_queue.clone();
+    let gcp = gcp.clone();
+    std::thread::spawn(move || {
+        // If indexer fails for whatever reason, let's spin it back up with exponential delay:
+        for i in 0..10 {
+            let options = options.clone();
+            let mpc_contract_id = mpc_contract_id.clone();
+            let account_id = account_id.clone();
+            let sign_queue = sign_queue.clone();
+            let gcp = gcp.clone();
+
+            // TODO/NOTE: currently indexer does not have any interrupt handlers and will never yield back
+            // as successful. We can add interrupt handlers in the future but this is not important right
+            // now since we managing nodes through integration tests that can kill it or through docker.
+            let Err(err) = indexer::run(options, mpc_contract_id, account_id, sign_queue, gcp)
+            else {
+                break;
+            };
+            tracing::error!(%err, "indexer failed");
+            std::thread::sleep(std::time::Duration::from_secs(2u64.pow(i)));
```

Comment:
Is sleeping `2**9=512` seconds too long?

Reply:
yeah that could be too long. Let's just cap it to something like 5 mins instead of 10mins. And instead of 10 loops, we can just loop infinitely